### DISCUSSION TITLE
Passing information between multiple sessions of an experiment

### DISCUSSION THREAD

**Discussion by Medha66** [jspsych/jsPsych#2611] [Q&A]
Hi everyone,

I'm running an experiment (coded in jsPsych 6.3.1 and hosted on cognition.run) in which participants have to complete tasks on two separate days. I have uploaded them as different experiments on cognition.run because each day involves different tasks. The stimulus is a 4x4 grid made up of images and I need to randomize the position of the stimuli on this grid across participants. The problem is that I need to make sure that the randomly chosen positions of the stimuli on the grid they see on day 1 are the same as what they see on day 2. I'm unable to figure out how to pass this information from day 1 to day 2. 

I tried creating different random sequences offline and giving the participant a unique ID (that identifies this sequence) that they have to enter on day 2 but I would prefer not having to rely on participants to get this information. Is there any way I can automatically save the stimulus sequence or some identifier on day 1 and access it on day 2?

Thanks,
Medha

**Comment by becky-gilbert** [comment]
Hi @Mehda66, I don't think there's an easy/automated way to do this on cognition.run. So what I might do if I were you is store an object in your 2nd session task that maps the participant identifier and condition. In the code, you can use the participant ID to select the version during that session (make sure you also have a default value in case something goes wrong). Then you'd need to manually update the object as soon as a participant finishes the 1st session.
Another way to do this is to have the version number passed into the task via a URL query parameter, but then you would need to distribute individualized links (based on their version) for your participants to use for their 2nd session.
I hope this helps! Let us know if you need further guidance.

**Comment by ChristopheBossens** [comment]
I have not used this myself, but could you use the [web storage API](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) to store local data in your first session? My idea would be to write some JavaScript code to store the unique ID in localStorage in the script of your first session, and then read from localStorage in the script of your second session.
